Templates/DefaultProject/Template/cmake/EngineFinder.cmake

```diff
@@ -47,46 +84,71 @@ Engine registration is required before configuring a project.
 Run 'scripts/o3de register --this-engine' from the engine root.
 ]=])
 
-# Read the ~/.o3de/o3de_manifest.json file and look through the 'engines_path' object.
-# Find a key that matches LY_ENGINE_NAME_TO_USE and use that as the engine path.
+# Read the ~/.o3de/o3de_manifest.json file and look through the 'engines' array.
+# Search all known engines for one with the matching name and compatible version
```

Comment:
General question...
Say my engines array looks like `["C:/EngineA", "C:/EngineB"]` and I'm working on a project that uses engine B.  It will always look at and read engine A's `engine.json` first.  Does this mean that if I happen to modify engine A's `engine.json` that it will cause ZERO_CHECK on the project (that uses engine B) to force another CMake configure?

Reply:
That's a good question, it is my understanding that unless you use `CMAKE_CONFIGURE_DEPENDS` to add an `engine.json` as a dependency, it will not be considered when checking.  In my testing, modifications to the other engine.json files did not trigger ZERO_CHECK

Reply:
Correct, the CMAKE_CONFIGURE_DEPENDS adds a file dependency for ZERO_CHECK.

The engine.json that is being used should have that `engine.json` added as a CMAKE_CONFIGURE_DEPENDS only